lib/array_equals.rb

```diff
@@ -1,5 +1,28 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  
+  match = true
+  array1.each_with_index do |value, index|
+    if array1[index] != array2[index]
+      match = false
+    end
+    return match
```

Comment:
This will cause the loop to exit after 1 iteration.  So if the 1st elements are the same it will return true.